### ISSUE TITLE
doc: Unclear Btrfs driver unefficiency

### ISSUE THREAD

**Issue by JoseFMP** [canonical/lxd#13688]
Document: `reference/storage_btrfs.md`

In the fragment:

```
This approach is not very efficient and might cause issues when creating snapshots.
```

It is not clear whether the `is not very efficient` refers to the whole Btrfs driver in general, or it refers only to `creating snapshots`. 

In other words it is not clear whether the word `and` or the word `when` is the top level connector and therefore by reading the sentence I cannot determine if the inefficiency is concerned only with snapshots or with the Brtfs driver in general.

---
Document: `reference/storage_btrfs.md`

**Comment by simondeziel** [comment]
> therefore by reading the sentence I cannot determine if the inefficiency is concerned only with snapshots or with the Brtfs driver in general.

While we can certainly improve the doc, the inefficiency with btrfs is centered around its lack of block device support and how it needs to be worked around. As such, when block devices are needed, LXD has to resort to using big files with Copy-on-Write (CoW) disabled. Because there is no CoW, snapshot'ing those files is costly in terms of time and space.

There's another downside to disabling CoW that we don't mention in the doc because it doesn't affect the default setup. If you've tuned your storage pool to use btrfs compression support, this also gets turned off for the big files backing block devices.

So in summary, btrfs' performance is subpar when snapshot'ing block devices (typically VMs) but it otherwise should be fine.
